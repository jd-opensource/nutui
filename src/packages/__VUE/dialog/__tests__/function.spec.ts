import DialogFunction from './dialog';

function sleep(delay = 0): Promise<void> {
  return new Promise((resolve) => {
    setTimeout(resolve, delay);
  });
}

describe('function Dialog', () => {
  test('show dialog info ', async () => {
    DialogFunction({
      title: 'title',
      content: 'content'
    });
    await sleep(1000);
    expect(document.querySelector('.nut-dialog')).toBeTruthy();
    let textTitle = document.querySelector('.nut-dialog .nut-dialog__header') as HTMLDivElement;
    expect(textTitle.innerHTML).toEqual('title');
    let content = document.querySelector('.nut-dialog .nut-dialog__content view') as HTMLDivElement;
    expect(content.innerHTML).toEqual('content');
  });
});
