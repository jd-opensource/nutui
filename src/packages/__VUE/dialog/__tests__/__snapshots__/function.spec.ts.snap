// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`function Dialog > dialog cancelText okText 1`] = `"<view class="nut-button__wrap"><!--v-if--><!--v-if--><view class="">取消文案自定义</view></view>"`;

exports[`function Dialog > dialog cancelText okText 2`] = `"<view class="nut-button__wrap"><!--v-if--><!--v-if--><view class="">确定文案自定义</view></view>"`;
